tests/function/test_config.py

```diff
@@ -614,3 +615,272 @@ def test_interval_in_virtwho_sysocnfig(self, virtwho, function_sysconfig):
                 and result['loop'] == 60)
 
         function_sysconfig.clean()
+
+    @pytest.mark.tier1
+    def test_pid_files_permission(self, virtwho, ssh_host):
+        """Test the sysconfig and pid files permission
+
+        :title: virt-who: config: test the sysconfig and pid files permission
+        :id: 21bf00ef-8b58-48b8-a36c-d65bc7b18dc0
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Stop virt-who to check virt-who sysconfig permission
+            2. Start virt-who to check virt-who.pid file permission
+
+        :expectedresults:
+
+            1. the permission of /etc/sysoncifg/virt-who file should be -rw-------
+            2. the permission of /var/run/virt-who.pid file should be  -rw-------
+        """
+        # check virt-who sysconfig file permission
+        virtwho.stop()
+        cmd = f"ls -l '{SYSCONFIG_FILE}'"
+        ret, output = ssh_host.runcmd(cmd)
+        assert (ret == 0
+                and output is not None
+                and output != ""
+                and "-rw-------" in output)
+
+        # check virt-who.pid file permission
+        pid_file = "/var/run/virt-who.pid"
+        virtwho.start()
+        cmd = f"ls -l '{pid_file}'"
+        ret, output = ssh_host.runcmd(cmd)
+        assert (ret == 0
+                and output is not None
+                and output != ""
+                and "-rw-------" in output)
+
+@pytest.mark.usefixtures('function_globalconf_clean')
+@pytest.mark.usefixtures('class_hypervisor')
+@pytest.mark.usefixtures('class_virtwho_d_conf_clean')
+class TestConfigurationNegative:
+    @pytest.mark.tier2
+    def test_debug_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the debug option in /etc/virtwho.conf
+
+        :title: virt-who: config: test debug negative option
+        :id: d06ee580-767f-43f6-9dab-51947ed9d4db
+        :caseimportance: High
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with "debug=null" in [global] section in /etc/virt-who.conf file
+        :expectedresults:
+
+            1. no [DEBUG] log printed
+        """
+        globalconf.update('global', 'debug', '')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['thread'] == 1
+                and result['error'] == 0
+                and result['debug'] is False)
+
+    @pytest.mark.tier2
+    def test_oneshot_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the oneshot option in /etc/virtwho.conf
+
+        :title: virt-who: config: test oneshot negative option
+        :id: 843f67cb-c98e-4074-9527-ae216fb036df
+        :caseimportance: High
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with "oneshot=True" in /etc/virt-who.conf
```

Comment:
should be `oneshout=[null value]`

Reply:
Yep, copied it from the positive case, forgot to update it :)